src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -1155,6 +1207,157 @@ private void validateStaticCodeAnalysisSettings(ProgrammingExercise programmingE
+                    ErrorKeys.INVALID_AUXILIARY_REPOSITORY_CHECKOUT_DIRECTORY);
+        }
+    }
+
+    private void validateAuxiliaryRepositoryCheckoutDirectoryLength(AuxiliaryRepository auxiliaryRepository) {
+        if (auxiliaryRepository.getCheckoutDirectory().length() > AuxiliaryRepository.MAX_CHECKOUT_DIRECTORY_LENGTH) {
+            throw new BadRequestAlertException("The checkout directory path '" + auxiliaryRepository.getCheckoutDirectory() + "' is too long!", AUX_REPO_ENTITY_NAME,
+                    ErrorKeys.INVALID_AUXILIARY_REPOSITORY_CHECKOUT_DIRECTORY);
+        }
+    }
+
+    private void validateAuxiliaryRepositoryCheckoutDirectoryDuplication(AuxiliaryRepository auxiliaryRepository, List<AuxiliaryRepository> otherRepositories) {
+        for (AuxiliaryRepository repo : otherRepositories) {
+            if (repo.getCheckoutDirectory() != null && repo.getCheckoutDirectory().equals(auxiliaryRepository.getCheckoutDirectory())) {
+                throw new BadRequestAlertException("The checkout directory path is already defined for another additional repository!", AUX_REPO_ENTITY_NAME,
+                        ErrorKeys.INVALID_AUXILIARY_REPOSITORY_CHECKOUT_DIRECTORY);
+            }
+        }
+    }
+
+    private void validateAuxiliaryRepositoryDescriptionLength(AuxiliaryRepository auxiliaryRepository) {
+        if (auxiliaryRepository.getDescription() != null && auxiliaryRepository.getDescription().length() > AuxiliaryRepository.MAX_DESCRIPTION_LENGTH) {
+            throw new BadRequestAlertException("The provided description is too long!", AUX_REPO_ENTITY_NAME, ErrorKeys.INVALID_AUXILIARY_REPOSITORY_DESCRIPTION);
+        }
+    }
+
+    private void validateAuxiliaryRepository(AuxiliaryRepository auxiliaryRepository, List<AuxiliaryRepository> otherRepositories) {
+
+        // Id of the auxiliary repository must not be set, because the id is set
+        // by the database.
+        validateAuxiliaryRepositoryId(auxiliaryRepository);
+
+        // We want to force the user to set a name of the auxiliary repository, otherwise we
+        // cannot determine which name we should use for setting up the repo on the VCS.
+        validateAuxiliaryRepositoryNameExists(auxiliaryRepository);
+
+        // The name must not be longer than 100 characters, since the database column is
+        // limited to 100 characters.
+        validateAuxiliaryRepositoryNameLength(auxiliaryRepository);
+
+        // We want to avoid using the same auxiliary repository name multiple times
+        validateAuxiliaryRepositoryNameDuplication(auxiliaryRepository, otherRepositories);
+
+        // The name must not match any of the names of the already present repositories, otherwise
+        // we get an undefined state.
+        // Currently, the names "exercise", "solution", and "tests" are restricted.
+        validateAuxiliaryRepositoryNameRestricted(auxiliaryRepository);
+
```

Comment:
Technically, we could also check against the repositories of all users (or the pattern) if they had one, but that will be too much, I think. I just wanted to mention it in case this is/gets important because such name conflicts would be possible as well.

Reply:
We discussed this internally already but came to the conclusion that this is the responsibility of the instructor to set this name appropriately. I guess we can add this in a follow-up PR.